lib/controllers/esri-token-auth.js

```diff
@@ -0,0 +1,86 @@
+/* jshint node: true, esnext: true */
+"use strict";
+var router = require('express').Router();
+var request = require('request');
+var bodyParser = require('body-parser');
+var url = require('url');
+
+module.exports = function(options) {
+    if (!options || !options.servers) {
+        return;
+    }
+
+    // The maximum size of the JSON data.
+    let postSizeLimit = options.postSizeLimit || '1024';
+
+    let tokenServers = parseUrls(options.servers);
+
+    router.use(bodyParser.json({limit:postSizeLimit, type:'application/json'}));
+    router.post('/', function(req, res, next) {
+        let parameters = req.body;
+
+        if (!parameters.url || parameters.url.length === 0) {
+            return res.status(400).send('No URL specified.');
+        }
+
+        let targetUrl = parseURL(parameters.url);
+        if (!targetUrl || targetUrl.length === 0 && (typeof targetUrl === string)) {
```

Comment:
This logic is not right.  `typeof targetUrl` returns `"string"`.  `string` (without quotes) is an undefined symbol and I believe this code will crash if `targetUrl.length` is 0.

Also I believe `&&` has higher precedence than `||`, so this is equivalent to:
```
if (!targetUrl || (targetUrl.length === 0 && typeof targetUrl === string)) {
```
which I don't think is what you intended.

Reply:
This is terrible code on my behalf... I don't know what I was doing, thanks for picking this up.

e874f7e2064db6fa592d0cc421a6d805a8a273da